packages/@react-spectrum/card/src/GalleryLayout.tsx

```diff
@@ -138,105 +138,107 @@ export class GalleryLayout<T> extends BaseLayout<T> {
     let y = this.margin;
     let availableWidth = visibleWidth - this.margin * 2;
 
-    // Compute aspect ratios for all of the items, and the total width if all items were on in a single row.
-    let ratios = [];
-    let totalWidth = 0;
-    let minRatio = this.minItemSize.width / this.minItemSize.height;
-    let maxRatio = availableWidth / this.minItemSize.height;
-
-    for (let node of this.collection) {
-      let ratio = node.props.width / node.props.height;
-      if (ratio < minRatio) {
-        ratio = minRatio;
-      } else if (ratio > maxRatio && ratio !== minRatio) {
-        ratio = maxRatio;
+    // If avaliable width is not greater than 0, skip node layout calculations
+    if (availableWidth > 0) {
```

Comment:
The change shown in git seems large, but this is literally just adding this wrapping if statement around the bulk of the `buildCollection` logic. Without this check, the ratios calculated are all out of wack and can lead to odd values (NaN) for the row height calculations. This in turn breaks the calculated `y` and `height` rect values that are propagated to OverscanManager, Virtualizer, etc and leads to errors for future `getVisibleLayoutInfos` calculations hence the Cards not reappearing when resizing the container